tests/tests.js

```diff
@@ -1957,4 +1957,244 @@ $(function() {
+        ok(false, msg);
+      }).on('success', function(data) {
+        user1_id = data["__id__"]
+        ok(true, msg);
+      }).on('complete', createUser2);
+    }
+
+    function createUser2() {
+      msg = "Create user2";
+      webservice.createUser({email: user2.email, password: user2.password}).on('error', function() {
+        ok(false, msg);
+      }).on('success', function(data) {
+        user2_id = data["__id__"]
+        ok(true, msg);
+      }).on('complete', loginUser1_CreateUserData);
+    }
+
+    function loginUser1_CreateUserData() {
+      webservice.login(user1).on('success', function(data) {
+        user1_session_token = data.session_token
+        ok(true, 'Logged in user1');
+      }).on('error', function() {
+        ok(false, 'Failed to login as user1');
+      }).on('complete', setUser1Value);
+    }
+
+    function setUser1Value() {
+      webservice.set(user_object_key, privateUserObj).on('success', function() {
+        ok(true, 'Successfully set value of user-level data while logged in as user1');
+      }).on('error', function() {
+        ok(false, 'Failed to set value of user-level data while logged in as user1');
+      }).on('complete', logoutUser1_CreateUserData);
+    }
+
+    function logoutUser1_CreateUserData() {
+      webservice.logout({email: user1.email, password: user1.password, session_token: user1_session_token}).on('success', function(data) {
+        ok(true, 'User1 logged out');
+      }).on('error', function() {
+        ok(false, 'Failed to logout as user1');
+      }).on('complete', loginUser2_NoACL);
+    }
+
+    function loginUser2_NoACL() {
+      webservice.login(user2).on('success', function(data) {
+        user2_session_token = data.session_token
+        ok(true, 'Logged in user2');
+      }).on('error', function() {
+        ok(false, 'Failed to login as user2');
+      }).on('complete', verifyNoACL);
+    }
+
+    function verifyNoACL() {
+      webservice.get(user_object_key, {applevel: false}).on('success', function() {
+        ok(false,'User2 had access to user1 data')
+      }).on('error', function() {
+        ok(true, 'User2 could not find the key');
+      }).on('complete', logoutUser2_NoACL);
+    }
+
+    function logoutUser2_NoACL() {
+      webservice.logout({email: user2.email, password: user2.password, session_token: user2_session_token}).on('success', function(data) {
+        ok(true, 'User2 logged out');
+      }).on('error', function() {
+        ok(false, 'Failed to logout as user2');
+      }).on('complete', loginUser1_CreateACL);
+    }
+
+    function loginUser1_CreateACL() {
+      webservice.login(user1).on('success', function(data) {
+        user1_session_token = data.session_token
+        ok(true, 'Logged in as user1');
+      }).on('error', function() {
+        ok(false, 'Failed to login as user1');
+      }).on('complete', createACL);
+    }
+
+    function createACL() {            
+      msg = "Create ACL as user1";
```

Comment:
Same message for success and failure. Do these messages display when the test is running?